ToolChain/Assembler/Assembler.py

```diff
@@ -23,70 +23,75 @@ def __init__(self, inputFile=None, outputFile=None):
+        """
+        This method assembles the binary data itself. It will parse each line individually and extract the info needed:
+        1.If the label flag is set to 0, it's a regular instruction, string, number, of memref.
+        2.If the label flag is set to 1, it's an internal symbol.
+        3.If the label flag is set to 2, it's an external symbol.
+        In cases 2 and 3, the symbols are added to their respective lists along with their memory location.
+        Memory locations are relative to the beginning of the file. Instructions and data are given set lengths.
+        Internal and external symbol lists contain tuples containing the symbol name, and its offset.
+        These will be used to fill the .o file information before the binary data.
+        :param inputFile: str, the name of the file to be read.
+        :param outputFile: str, the name of the file to be written to.
+        :return:
+        """
+
+        parser = Parser()                    # Parser object which will return the binary code for instructions
+        masterString = b""                   # String that will eventually be our output file
+        xmlstring = b""                      # String that will contain the "xml" data at the beginning of the .o file
+        global_list = []                     # Temporary list to hold the names of global(external) symbols
+        externalList = []                    # Master list of all external symbols to be used by linker
+        internalList = []                    # Master list of all internal symbols
+        offset = 0                           # Offset used to calculate a label's offset from the beginning of the file
 
         file = open(inputFile, mode="r")
-        file_lines = file.readlines()
+        fileLines = file.readlines()
         file.close()
 
-        # This block assembles the binary data itself. It will parse each line individually and extract the info needed:
-        # 1.If the label flag is set to 0, it's a regular instruction, string, number, of memref.
-        # 2.If the label flag is set to 1, it's an internal symbol.
-        # 3.If the label flag is set to 2, it's an external symbol.
-        # In cases 2 and 3, the symbols are added to their respective lists along with their memory location.
-        # Memory locations are relative to the beginning of the file. Instructions and data are given set lengths.
-        # Internal and external symbol lists contain tuples containing the symbol name, and its offset.
-        # These will be used to fill the .o file information before the binary data.
-
         masterString += b"<Text>"
 
-        for line in file_lines:
-
+        for line in fileLines:
+            # If the file isn't empty, process the content
             if line[0] != "\n":
+                # We simply ensure that there is something on the line, not just newline (empty)
+                instruction, offset, label_flag = parser.parse(line)
```

Comment:
maybe rename `label_flag` to `instructionType`? Or just make it camelCase?

Reply:
Please rename variable